src/content-script/listeners/player.ts

```diff
@@ -47,40 +49,68 @@ function findVideoPlayer(): HTMLVideoElement | null {
+  // TODO: Once implemented, include communism/facism check
+  function sendVideoEvent(socketEvent: VideoSocketEvents, hostRequired = true) {
     return (event: any) => {
       const state = store.getState();
-      const { isHost } = state.party;
-      const { id: partyId } = state.party;
-      const { currentTime, duration, playbackRate } = event.target;
+      const { isHost, hash } = state.party;
 
-      const payload = { partyId, currentTime, duration, playbackRate };
+      // if party not created, don't send anything
+      if (!hash) return;
 
+      // construct payload
+      const { paused, currentTime, duration, playbackRate } = event.target;
+      const eventData: VideoEventData = { eventType: socketEvent, isHost, paused, currentTime, playbackRate, duration };
+      const payload: VideoEvent = { partyHash: hash, eventData };
+
+      // send payload
       if (hostRequired && isHost) {
         debug(`Host emitting ${socketEvent}`);
-        socket.emit(socketEvent, payload);
+        socket.emit(SocketEvents.VIDEO_EVENT, payload);
       } else if (!hostRequired) {
         debug(`User emitting ${socketEvent}`);
-        socket.emit(socketEvent, payload);
+        socket.emit(SocketEvents.VIDEO_EVENT, payload);
       }
     };
   }
 
-  player.onplay = sendVideoEvent(SocketEvents.VIDEO_PLAY);
-  player.onpause = sendVideoEvent(SocketEvents.VIDEO_PAUSE);
-  player.onseeked = sendVideoEvent(SocketEvents.VIDEO_SEEKED);
-  player.onprogress = sendVideoEvent(SocketEvents.VIDEO_PROGRESS, false);
+  player.onplay = sendVideoEvent(VideoSocketEvents.VIDEO_PLAY);
+  player.onpause = sendVideoEvent(VideoSocketEvents.VIDEO_PAUSE);
+  player.onseeked = sendVideoEvent(VideoSocketEvents.VIDEO_SEEKED);
+  player.onprogress = sendVideoEvent(VideoSocketEvents.VIDEO_PROGRESS, false);
+}
+
+//------------------------------------------------------------------------------
+
+function addVideoSocketListeners(player: HTMLVideoElement) {
+  socket.on(SocketEvents.VIDEO_EVENT, (data: VideoEventData) => {
+    // TODO: Once implemented, include communism/facism check
+    if (data.isHost) {
+      // match player play/pause state
+      if (data.paused) player.pause();
+      else player.play();
+
+      // match player playback speed
+      player.playbackRate = data.playbackRate;
+
+      // resync if needed and not host. handles seek events
+      if (Math.abs(player.currentTime - data.currentTime) > MAX_DESYNC_SEC) {
+        player.currentTime = data.currentTime;
+      }
+    }
+  });
 }
 
 //------------------------------------------------------------------------------
 
 function attachToVideoPlayer() {
   const videoPlayer = findVideoPlayer();
-  if (videoPlayer) addVideoListeners(videoPlayer);
+  if (videoPlayer) {
+    addVideoListeners(videoPlayer);
+    addVideoSocketListeners(videoPlayer);
+  }
 }
 
 //------------------------------------------------------------------------------
 
 attachToVideoPlayer();
```

Comment:
Not sure what approaches you took for testing, but this wasn't working for me. The problem here is that the content script is injected when we navigate to youtube.com. We need to re-attach on any youtube page change because otherwise `player` will be `null` and we won't send any events.